app/upgrade_handler.go

```diff
@@ -23,19 +24,31 @@ import (
 )
 
 const (
-	tmp = "tmp"
+	upgradeName = "v2.5.0"
 )
 
-// TODO: rename upgrade title
-func (app ShentuApp) setTmpUpgradeHandler() {
+func (app ShentuApp) setUpgradeHandler() {
 	app.UpgradeKeeper.SetUpgradeHandler(
-		tmp,
+		upgradeName,
 		func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
-			migrationOrder := make([]string, len(fromVM))
-			i := 0
+			migrationOrder := make([]string, 0, len(fromVM))
+			hasParams, hasICA := false, false
 			for moduleName := range fromVM {
-				migrationOrder[i] = moduleName
-				i++
+				if moduleName == crisistypes.ModuleName {
+					continue
+				} else if moduleName == paramstypes.ModuleName {
+					hasParams = true
+				} else if moduleName == icatypes.ModuleName {
+					hasICA = true
+				}
+				migrationOrder = append(migrationOrder, moduleName)
+			}
+			//to satisfy the assertNoForgottenModules of SetOrderMigrations
+			if !hasParams {
```

Comment:
Shouldn't be `if hasParams` here?

Reply:
if params is already in `fromVM`, it should be avoided to be added again.